### ISSUE TITLE
desktop menus

### ISSUE THREAD

**Issue by gedw99** [dawenxi-tech/2fa#12]
Just a thought for later, but I think can adapt this to have menus for gio desktop apps later too.

The same mechanism that shows a menu in the tray can be adapted to show menus for a gio desktop app perhaps. Lets see when we get to it.

**Comment by heramerom** [comment]
Yeah, I was a iOS developer many years ago and has a little macos dev experience.

In macos or ios develop, an application is a singleton instance, it can only has one delegate. When systray runs, it will set a delegate and when gio runs, it also set a delegate, that conflict. So it will run into mess some time.

And in my case, I was modify Gio source code to add a GioAppDelegate property and set property value later use `NSObject performSelector:withObject: ` api.
``` objective-c
@interface GioAppDelegate : NSObject<NSApplicationDelegate>
@property (nonatomic, strong) NSStatusItem *statusItem;
@end
```
```objective-c

[delegate performSelector:@selector(setStatusItem:) withObject:statusItem];

```

In fact, we can even do not change Gio source code, That is use the `class_addProperty` Api, https://stackoverflow.com/questions/7819092/how-can-i-add-properties-to-an-object-at-runtime, but that was not suggest by Apple.

Btw, the statusItem must as a delegate's property so it will appear in menu bar. That cost me a lot of time to find out reason?
https://stackoverflow.com/a/54773268
